kratos/input_output/stl_io.cpp

```diff
@@ -78,32 +78,29 @@ void StlIO::WriteModelPart(const ModelPart & rThisModelPart)
 template<class TContainerType>
 void StlIO::WriteEntityBlock(const TContainerType& rThisEntities)
 {
+    std::size_t num_degenerate_geometries = 0;
     for (auto & r_entity : rThisEntities) {
         const auto & r_geometry = r_entity.GetGeometry();
-
-        // restrict to triangles only for now
-        const bool is_triangle = (
-            r_geometry.GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Triangle3D3 ||
-            r_geometry.GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Triangle3D6);
-
-        if (is_triangle) {
+        if (IsValidGeometry(r_geometry, num_degenerate_geometries)) {
             WriteFacet(r_geometry);
         }
     }
+    KRATOS_WARNING_IF("STL-IO", num_degenerate_geometries > 0) 
```

Comment:
I think using std::endl for warnings is a fair usage of flush

Reply:
Ok, done!